specs/linux/shadow.table

```diff
@@ -0,0 +1,18 @@
+table_name("shadow")
+description("Local system users encrypted passwords and related information.")
+schema([
+    Column("password_status", TEXT, "Password status"),
+    Column("hash_alg", TEXT, "Password hashing algorithm"),
+    Column("last_change", BIGINT, "Date of last password change (starting from UNIX epoch date)"),
+    Column("min", BIGINT, "Minimal number of days between password changes"),
+    Column("max", BIGINT, "Maximum number of days between password changes"),
+    Column("warning", BIGINT, "Number of days before password expires to warn user about it"),
+    Column("inactive", BIGINT, "Number of days after password expires until account is blocked"),
+    Column("expire", BIGINT, "Number of days since UNIX epoch date until account is disabled"),
+    Column("flag", BIGINT, "Reserved"),
+    Column("username", TEXT, "Username"),
```

Comment:
Hey @reedloden, sorry I didn't catch this before. But since you're using `username` in the query constraints can you place a kwarg in the `username` `Column` definition, `index=True`. This will define the table with `username` as a primary key and prevent `WHERE username = ...` from being optimized out of the query.

Reply:
Done! This is the first table that I see that uses `username` as a primary key, so if this is indeed a problem, I bet several other tables need to be fixed as well (for those who don't use something else as a primary key).